src/java/org/apache/cassandra/service/MigrationManager.java

```diff
@@ -188,6 +219,25 @@ public static boolean shouldPullSchemaFrom(InetAddress endpoint)
                 && !Gossiper.instance.isGossipOnlyMember(endpoint);
     }
 
+    public static boolean shouldPullSchemaFrom(InetAddress endpoint, UUID theirVersion)
+    {
+        /*
+         * Don't request schema from nodes with a differnt or unknonw major version (may have incompatible schema)
```

Comment:
nit typos in this + the original comment

Reply:
Fixed